tsl/src/nodes/decompress_chunk/planner.c

```diff
@@ -378,6 +391,90 @@ find_attr_pos_in_tlist(List *targetlist, AttrNumber pos)
 	pg_unreachable();
 }
 
+static bool
+qual_is_vectorizable(DecompressChunkPath *path, Node *qual)
+{
+	/* Only simple "Var op Const" binary predicates for now. */
+	if (!IsA(qual, OpExpr))
+	{
+		return false;
+	}
+
+	OpExpr *o = castNode(OpExpr, qual);
+
+	if (list_length(o->args) != 2)
+	{
+		return false;
+	}
+
+	if (IsA(lsecond(o->args), Var) && IsA(linitial(o->args), Const))
+	{
+		/* Try to commute the operator if the constant is on the right. */
+		Oid commutator_opno = get_commutator(o->opno);
+		if (OidIsValid(commutator_opno))
+		{
+			o->opno = commutator_opno;
+			o->opfuncid = InvalidOid;
```

Comment:
why does this need to be `InvalidOid` instead of `get_opcode(commutator_opno);`

Reply:
I am also not sure about the `opfuncid`. However, in `CommuteOpExpr` the `opfuncid` is also set to `InvalidOid`.

Reply:
Right, I think there's no particular reason, just repeating the CommuteOpExpr. I added a comment about this.